File: android/src/main/kotlin/com/example/zendesk_sdk/models/ZendeskTicketRequest.kt
package com.example.zendesk_sdk.models

import com.example.zendesk_sdk.ZendeskSdkChannel
import io.flutter.plugin.common.MethodCall

/// Ticket submission context combining [user], [tripId], and optional fields.
data class ZendeskTicketRequest(
    val user: ZendeskUser,
    val tripId: String,
    val customFields: List<ZendeskCustomField> = emptyList(),
) {
    companion object {
        fun fromMethodCall(call: MethodCall): ZendeskTicketRequest? {
            val user = ZendeskUser.fromMethodCall(call)
            val tripId = call.argument<String>(ZendeskSdkChannel.Argument.TRIP_ID) ?: ""
            if (user.userId.isEmpty() || tripId.isEmpty()) {
                return null
            }
            return ZendeskTicketRequest(
                user = user,
                tripId = tripId,
                customFields = ZendeskCustomField.listFromMethodCall(call),
            )
        }
    }
}

File: ios/zendesk_sdk/Sources/zendesk_sdk/Models/ZendeskTicketRequest.swift
import Flutter

/// Ticket submission context combining [user], [tripId], and optional fields.
struct ZendeskTicketRequest {
    let user: ZendeskUser
    let tripId: String
    let customFields: [ZendeskCustomFieldModel]

    static func from(call: FlutterMethodCall) -> ZendeskTicketRequest? {
        guard let args = call.arguments as? [String: Any],
              let tripId = args[ZendeskSdkChannel.Argument.tripId] as? String,
              !tripId.isEmpty
        else {
            return nil
        }

        let user = ZendeskUser.from(call: call)
        if user.userId.isEmpty {
            return nil
        }

        return ZendeskTicketRequest(
            user: user,
            tripId: tripId,
            customFields: ZendeskCustomFieldModel.list(from: args)
        )
    }
}
